tests/unit/cmd/scan/test_ci.py

```diff
@@ -121,19 +127,15 @@ def test_ci_cmd_uses_right_mode_header(
     get_list_mock: Mock,
     scan_commit_range_mock: Mock,
     cli_fs_runner: click.testing.CliRunner,
-    monkeypatch,
     env: Dict[str, str],
     expected_mode: str,
 ):
-    # GIVEN a CI env
-    for key, value in env.items():
-        monkeypatch.setenv(key, value)
-
     get_list_mock.return_value = ["a"] * 51
     scan_commit_range_mock.return_value = 0
 
     # WHEN `secret scan ci` is called
-    result = cli_fs_runner.invoke(cli, ["secret", "scan", "ci"])
+    with patch("os.getenv", lambda k, default=None: env.get(k, default)):
+        result = cli_fs_runner.invoke(cli, ["secret", "scan", "ci"])
```

Comment:
Here, the patch is necessary. It seems like the `env` kwarg only overrides the keys that are passed, meaning that all the other env vars from the GitHub CI are still available and interfere with the test.

Reply:
Oh, interesting. It worked-on-my-machine but I did not have the GitHub env setup. In this case you can use the same approach as the other patch to avoid the lambda.